diff --git a/clar2wasm/build.rs b/clar2wasm/build.rs
index ea6c8b7..9724360 100644
--- a/clar2wasm/build.rs
+++ b/clar2wasm/build.rs
@@ -1,10 +1,15 @@
 /// Generate the standard library as a Wasm binary from the WAT source.
 #[allow(clippy::expect_used)]
 fn main() {
+    let out_dir = std::env::var("OUT_DIR").expect("OUT_DIR not set");
+    let wasm_path = std::path::Path::new(&out_dir).join("standard.wasm");
+
     match wat::parse_file("src/standard/standard.wat") {
         Ok(binary) => {
-            std::fs::write("src/standard/standard.wasm", binary)
+            std::fs::write(&wasm_path, binary)
                 .expect("Failed to write standard library");
+
+            println!("cargo:rustc-env=STANDARD_WASM_PATH={}", wasm_path.display());
         }
         Err(error) => {
             panic!("Failed to parse standard library: {error}");
